internal/nginx/extensions.go

```diff
@@ -62,6 +70,29 @@ func validateHashLBMethod(method string) (string, error) {
 	return "", fmt.Errorf("Invalid load balancing method: %q", method)
 }
 
+func validateRandomLBMethod(method string) (string, error) {
+	keyWords := strings.Split(method, " ")
+	if keyWords[0] == "random" {
+		if len(keyWords) == 1 || len(keyWords) == 2 && keyWords[1] == "two" ||
+			len(keyWords) == 3 && keyWords[1] == "two" && keyWords[2] == "least_conn" {
+			return method, nil
+		}
+	}
+	return "", fmt.Errorf("Invalid load balancing method: %q", method)
+}
+
+func validateRandomLBMethodForPlus(method string) (string, error) {
+	keyWords := strings.Split(method, " ")
+	if keyWords[0] == "random" {
+		if len(keyWords) == 1 || len(keyWords) == 2 && keyWords[1] == "two" ||
```

Comment:
since the total number of combinations for the random algorithms is small, I suggest to putting them in the `nginxPlusLBValidInput` map. This will simplify the code.